addons/account_voucher/migrations/9.0.1.0/post-migration.py

```diff
@@ -85,13 +85,19 @@ def create_payments_from_vouchers(env):
     )
     # Also recreate link from invoice to payment
     env.cr.execute(
-        """\
+        """
         INSERT INTO account_invoice_payment_rel (payment_id, invoice_id)
-        SELECT aml.payment_id, ai.id
-        FROM account_move_line aml
-        JOIN account_invoice ai ON aml.move_id = ai.move_id
-        WHERE NOT aml.payment_id IS NULL
-        GROUP BY aml.payment_id, ai.id
+        SELECT av.id as payment_id, ai.id as invoice_id
+        FROM account_invoice ai
+        JOIN account_move_line amli ON ai.move_id = amli.move_id
+        JOIN account_partial_reconcile apr ON 
+             (apr.credit_move_id = amli.id OR apr.debit_move_id = amli.id)
+        JOIN account_move_line amlp ON
+             (apr.credit_move_id = amlp.id OR apr.debit_move_id = amlp.id)
+        JOIN account_voucher av ON av.move_id = amlp.move_id
+        WHERE av.voucher_type IN ('sale', 'purchase')
```

Comment:
Why did you change this to sale or purchase type vouchers? Only vouchers with 'receipt' or 'payment' are migrated to account.payment in the first place. type https://github.com/OCA/OpenUpgrade/blob/9.0/addons/account_voucher/migrations/9.0.1.0/post-migration.py#L55

Reply:
I'm still reviewing. It came out at the time the post-migration script was executed that the voucher_type was already updated to the selection values for v9. In that case the condition is useless because it's fetching all vouchers of all types. What I don't know is why the voucher_type column was updated when there is no script for that.

Reply:
Any news on this?